elasticsearch/docker/templates/Dockerfile.j2

```diff
@@ -50,6 +53,38 @@ COPY --chown=1000:0 {{ '%s/' % artifacts_dir }} {{ artifacts_dir }}
 {% endif %}
 
 WORKDIR /usr/share/elasticsearch
+##Compile the C-library for kNN
+RUN set -ex; \
+ \
+    git clone https://github.com/nmslib/nmslib.git /usr/share/elasticsearch/nmslib; \
+ \
+    cd /usr/share/elasticsearch/nmslib \
+ \
+    && git checkout tags/v1.7.3.6 \
+ \
+    && cd similarity_search \
+ \
+    && cmake . \
+ \
+    && make \
+ \
+    && git clone https://github.com/opendistro-for-elasticsearch/k-NN.git /usr/share/elasticsearch/k-NN \
+ \
+    && cd /usr/share/elasticsearch/k-NN \
+ \
+    && git checkout v1.4.0.0 \
```

Comment:
Can i remove the tag checkout step and work with master branch?

Reply:
Sure. I would recommend the same procedure as other plugin